### ISSUE TITLE
Limit bandwidth or I/O

### ISSUE THREAD

**Issue by Rikerutz** [masc3d/btrfs-sxbackup#43]
Hello everybody,

We are running an application that is very I/O intensive. When pulling the first snapshots, there are moments when the VM freezes and btrfs-sxbackup needs to be stopped. The back-up is not to blame for this, but it adds just enough to freeze the server.
Is there any way to limit I/O or bandwidth so the back-up can be pulled at a slower pace?

**Comment by masc3d** [comment]
yes, you can use `ionice` to achieve this. here's an example cron entry for running on lowest io prio.
```
10 *		* * *		root	ionice -c3 -n7 btrfs-sxbackup -q run /mnt/backup/snapshots/mina-home -m
```
Btrfs has quite a few areas where it seems locking could be improved, so your issues may also relate to that. Would be interesteing to know  if `ionice` helped in your case.

**Comment by masc3d** [comment]
thing is, the io class is not passed on to remote shell commands. this is resolved by cadc7e76. io prio is now always idle for all btrfs send/receive invocations. `ionice -c3` is no longer necessary.
